m4/ax_check_libsemigroup.m4

```diff
@@ -11,17 +11,19 @@ AC_DEFUN([AX_CHECK_LIBSEMIGROUPS], [
   REQUI_LIBSEMIGROUPS_VERSION="$(cat .LIBSEMIGROUPS_VERSION)"
   need_included_libsemigroups=yes
   if test "$with_external_libsemigroups" = yes;  then
-	PKG_CHECK_MODULES([LIBSEMIGROUPS], [libsemigroups >= $REQUI_LIBSEMIGROUPS_VERSION],
-	[need_included_libsemigroups=no],
-	[need_included_libsemigroups=yes])
+	m4_ifdef([PKG_CHECK_MODULES], [
+	PKG_CHECK_MODULES([LIBSEMIGROUPS], 
+                          [libsemigroups >= $REQUI_LIBSEMIGROUPS_VERSION],
+			  [need_included_libsemigroups=no],
+			  [need_included_libsemigroups=yes])],
+	[AC_MSG_NOTICE([ignoring flag --with-external-libsemigroups, pkg-config is not available])])
```

Comment:
This error message is not strictly correct. This merely checks that the pkg.m4 macros are present. Yet, if they are present, one probably would also want to check that `pkg-config` itself is there and working. This can be done with PKG_PROG_PKG_CONFIG() [macro](https://people.freedesktop.org/~dbn/pkg-config-guide.html#using) and error out, or  print a warning, if it's not the case.

Reply:
The error message should say `Semigroups ./configure was created on a system without m4 macros for pkg-config available.`

Otherwise, in particular if one distributes a tarball built on such a system, its users, with all the pkg-config tools present, will be misled by the present message.

Reply:
Yup, agreed.